vertical-pod-autoscaler/docs/proposals/crash_loop_handling/README.md

```diff
@@ -0,0 +1,196 @@
+  4. This will ensure handling the spike as it is dependent on the current usage than a hysteresis of resource usage pattern and doubling will react exponentially to a potentially exponential spike in resouce usage.
+  5. We consider the following 3 variables (temporary names) threshold_no_of_crashes, last_crash_count_recorded_time, time_length_for_crash_check.
+  6. `threshold_no_of_crashes` talk about value being set during the initiation of recommender to threshold number of restarts due to crash beyond which recommender will scale up both CPU and memory. (Will have a default mostly as 3)
+  7. `last_crash_count_recorded_time` indicates the last crash count time recorded on the node. (Default: 0)
+  8. `time_length_for_crash_check` is `current machine time - last_crash_count_recorded_time`. (Default: 0)
+  9. We can set threshold_no_of_crashes to 3 as the default value. If there is a resource crunch even after bumping one of CPU or memory based on current usage and has resulted in a crash, threshold_no_of_crashes serves a threshold beyond which both resources are to be scaled up irrespective of current usage consumption.
+  10. The variables `threshold_no_of_crashes, last_crash_count_recorded_time, time_length_for_crash_check` are reset to default if the VPA recommender itself restarts. Example: Let's say we recommended the memory doubling but had constant crashes for more than 3 times. Then we get into else condition and double both memory and CPU irrespective of whatever the recommendation for individual resource is.
+
+#### Scale Down:
+Scale down to a variable between 1 and 2 times the local maxima within a time window
+  1. Two new variables called scale_down_safety_margin (default to 1.2), scale_down_monitor_time_window (default to 1 hour) are used.
+  2. We find the local maxima w.r.t resource usage within scale_down_monitor_time_window.
+  3. We only scale down to `scale_down_safety_margin` times the maximum resource usage recorded within `scale_down_monitor_time_window`.
+  4. `1 < scale_down_safety_margin < 2` (defaulted to 1.2). This ensures we always scale down to, slightly higher than previously seen highest resource usage within the time window but also lesser than scale up limit of doubling.
+  Note: It could also be given as a best practise to ensure `1 < scale_down_mutliple < 1.5` so that we do not over provision the resources while scaling down. As the parameter scale_down_safety_margin is configurable between 1 and 2, there is room for trial and error and eventually arrive at a custom combination of parmater values for each component based on their usage of VPA.
+  5. It also is crucial to scale down stepwise with shorter time windows. This ensures:
```

Comment:
Does this mean, that pod evictions will occur even more frequently because of scale-down actions?
That sounds quite dangerous for vertically scaling Pods that have Volumes attached (e.g. prometheus and loki in the Shoot control plane).
Also I'm a bit afraid, that scaling down so conservatively will also lead to a decrease node utilization again.

Reply:
I probably missed writing the "thresholdScaleDown" factor here in the proposal. I shall add that. The stepwise scale down (similar to current VPA histogram approach for decaying during scale down recommendation) here would refer to 2 parameters. ThresholdScaleDown parameter and also the ScaleDowntimeWindow. Both these ensure we don't end up evicting pods rapidly. We should not really scale down if the current usage > thresholdScaleDown * requests. This ensures we should probably live with whatever configuration we are having right now and not evict. However, if usage goes below threshold of requests then we initiate a scale down and there by eviction.